### ISSUE TITLE
[Bug] AI inline suggestions triggering despite AI features being disabled

### ISSUE THREAD

**Issue by mdmulligan** [streetsidesoftware/vscode-spell-checker#4878]
Type: <b>Bug</b>

1. Set `chat.disableAIFeatures` to `true`.
2. Edit the below included file, specifically the line that reads:
```
"lint:fix": "pnpm run eslint:fix && pnpm run prettier:fix && pnpm run cspell",
```
Attempt to add a `:` after the `cspell` prior to the `",`.
3. Observe what happens.

## Actual Results ##

* Inline suggestions are triggered.

## Expected Results ##

* AI features are **disabled** as the setting says they should be.

## Additional Details ##

Output of the `Developer: Extract Inline Suggest State` command:

```
{
    "name": "webcraft",
    "description": "A simple Minecraft clone built on web technologies.",
    "version": "0.0.0",
    "license": "SEE LICENSE IN LICENSE.md",
    "private": true,
    "author": {
        "name": "G'lek Tarssza",
        "email": "glek@glektarssza.com",
        "url": "https://glektarssza.com"
    },
    "homepage": "https://glektarssza.github.io/webcraft/",
    "bugs": {
        "url": "https://github.com/glektarssza/webcraft/issues"
    },
    "repository": {
        "type": "git",
        "url": "https://github.com/glektarssza/webcraft.git",
        "directory": "./app/"
    },
    "packageManager": "pnpm@10.22.0+sha512.bf049efe995b28f527fd2b41ae0474ce29186f7edcb3bf545087bd61fbbebb2bf75362d1307fda09c2d288e1e499787ac12d4fcb617a974718a6051f2eee741c",
    "type": "module",
    "scripts": {
        "dev": "vite serve --mode development",
        "test": "vitest --run --coverage",
        "test:ci": "vitest --run --coverage",
        "test:ui": "vitest --ui",
        "lint": "pnpm run eslint && pnpm run prettier",
        "lint:fix": "pnpm run eslint:fix && pnpm run prettier:fix && pnpm run cspell:",
        "lint:ci": "pnpm run eslint:ci && pnpm run prettier:ci && pnpm run cspell:ci",
        "cspell": "cspell lint --cache .",
        "cspell:ci": "cspell lint --no-cache .",
        "eslint": "eslint --cache ./src/ ./tests/",
        "eslint:fix": "eslint --cache --fix ./src/ ./tests/",
        "eslint:ci": "eslint ./src/ ./tests/",
        "prettier": "prettier --config ../.prettierrc.cjs --ignore-path ../.prettierignore --cache --check .",
        "prettier:fix": "prettier --config ../.prettierrc.cjs --ignore-path ../.prettierignore --cache --write .",
        "prettier:ci": "prettier --config ../.prettierrc.cjs --ignore-path ../.prettierignore --check .",
        "build": "pnpm run build:prod",
        "build:all": "pnpm run build:production && pnpm run build:development",
        "build:prod": "pnpm run build:production",
        "build:production": "vite build --mode production",
        "build:dev": "pnpm run build:development",
        "build:development": "vite build --mode development",
        "clean": "pnpm run clean:prod",
        "clean:all": "pnpm run clean:production && pnpm run clean:development",
        "clean:prod": "pnpm run clean:production",
        "clean:production": "del-cli \"./dist/\"",
        "clean:dev": "pnpm run clean:dev",
        "clean:development": "del-cli \"./dist/\"",
        "rebuild": "pnpm run rebuild:prod",
        "rebuild:all": "pnpm run clean:all && pnpm run build:all",
        "rebuild:prod": "pnpm run clean:prod && pnpm run build:prod",
        "rebuild:dev": "pnpm run clean:dev && pnpm run build:dev"
    },
    "devDependencies": {
        "@eslint/js": "^9.38.0",
        "@faker-js/faker": "^10.1.0",
        "@rollup/plugin-replace": "^6.0.2",
        "@types/node": "^24.9.1",
        "@vitest/browser": "^4.0.4",
        "@vitest/browser-playwright": "^4.0.4",
        "@vitest/coverage-istanbul": "^4.0.4",
        "@vitest/ui": "^4.0.4",
        "del-cli": "^7.0.0",
        "eslint": "^9.38.0",
        "eslint-config-prettier": "^10.1.8",
        "eslint-plugin-tsdoc": "^0.4.0",
        "playwright": "^1.56.1",
        "prettier": "^3.6.2",
        "sass": "^1.93.2",
        "ts-node": "^10.9.2",
        "typescript": "^5.9.3",
        "typescript-eslint": "8.46.2",
        "vite": "^7.1.12",
        "vitest": "^4.0.4"
    }
}


// <json>
// {
//     "inlineCompletion": {
//         "insertText": "words",
//         "range": {
//             "startLineNumber": 29,
//             "startColumn": 86,
//             "endLineNumber": 29,
//             "endColumn": 86
//         },
//         "idx": 0,
//         "completeBracketPairs": false,
//         "isInlineEdit": false,
//         "showInlineEditMenu": false,
//         "suggestionId": "sgt-2edefabc-6e59-42dc-98aa-25241156d07b"
//     }
// }
// </json>
```

Extension version: 4.3.0
VS Code version: Code - Insiders 1.107.0-insider (769ef65cd62212cde4db47c538acfea3063f1277, 2025-11-13T14:43:43.333Z)
OS version: Linux x64 6.17.7-arch1-2
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 5 5600H with Radeon Graphics (12 x 3808)|
|GPU Status|2d_canvas: enabled<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>trees_in_viz: disabled_off<br>video_decode: enabled<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off<br>webnn: disabled_off|
|Load (avg)|1, 1, 1|
|Memory (System)|27.26GB (17.98GB free)|
|Process Argv|/home/glek/Development/Repositories/GitHub/glektarssza/webcraft-v2 --crash-reporter-id 4d1296bd-f739-474e-a1de-184344146899|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|xfce|
|XDG_CURRENT_DESKTOP|XFCE|
|XDG_SESSION_DESKTOP|xfce|
|XDG_SESSION_TYPE|x11|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
pythonvspyt551cf:31249598
nativeloc1:31118317
dwcopilot:31158714
dwoutputs:31242946
copilot_t_ci:31333650
g012b348:31231168
6gi0g917:31259950
pythonrdcb7:31268811
pythonpcpt1cf:31399617
6518g693:31302842
4f60g487:31327383
onetestforazureexp:31335613
6abeh943:31336334
envsactivate1:31349248
editstats-enabled:31346256
aa_t:31379596
t123456:31413567
cloudbuttont:31366566
aihoversummaries_t:31413033
todos-1:31405333
qwen_all_req:31396049
3efgi100_wstrepl:31403338
trigger-command-fix:31379601
auto_model_enabled:31385282
use-responses-api:31390341
auto_g2946222:31414244
je187915:31407605
j1f0b290:31405249
3i15c187:31405337
control_hh9fb551:31411652
agentic_ask_h7d75428:31406124
ff8f5884:31411053
none:31413573
haab2303:31418848
chatvisible:31417128
expandedtokens:31418009
a9239246:31418438

```

</details>

**Comment by glektarssza** [comment]
Commenting from my main account (accidentally submitted this issue from my secondary account); this may very well be a VSCode issue. If so let me know and I'll refile an issue with them.

**Comment by Jason3S** [comment]
@glektarssza,

Cool project.

The spell checker extension will offer inline completion after it see `cspell:`. `cspell:` is the beginning of the inline directives for `cspell`, like `cspell:disable-next-line` to tell it to not check the next line. The extension offers to complete the directive.

I'm not sure what you are seeing. This is what happens to me:

<img width="522" height="69" alt="Image" src="https://github.com/user-attachments/assets/9ce66141-c1bb-40a7-b223-9d06ba65a64a" />

I could add an option to turn this off. I'm just not sure how many people will find the option.

**Comment by glektarssza** [comment]
> Cool project.

Thank you!

> The spell checker extension will offer inline completion after it see `cspell:`.

Okay, that does make sense. It's weird. The reason I thought it was to do with VSCode's AI features is because whether it was showing up or not was reacting to having the `inlineChat.enableV2` setting which appears to be an AI-related flag.

> I'm not sure what you are seeing. This is what happens to me:

This is what I am seeing (though in a very different font, that's likely a misconfigured editor setting somewhere).

> I could add an option to turn this off. I'm just not sure how many people will find the option.

I wonder if there's a way to improve the way the inline directive detection tells when it should apply or not? I'm not sure the conditions under which it's supposed to apply (e.g. comments only) and I get that trying to account for all possible language comment styles would be massively annoying. Only applying inside comments would resolve the issue, to be sure. Maybe looking at another extension such as [Better Comments](https://marketplace.visualstudio.com/items?itemName=aaron-bond.better-comments) for inspiration on how they handled detecting comments might be worthwhile?

Failing that, I think a good middle ground might be a setting to disable inline suggestions for files matching a globbing pattern rather than just disabling it globally? This is a feature I'd still want to leverage as I can definitely see value in it but I don't really want it triggering in files like `package.json`.
